404.html

```diff
@@ -11,39 +11,127 @@
   <meta property="og:title" content="Page not found">
   <script src="/scripts/scripts.js" type="module" crossorigin="use-credentials"></script>
   <script type="module">
-    import { sampleRUM } from '/scripts/lib-franklin.js';
-
-    window.addEventListener('load', () => {
-      if (document.referrer) {
-        const { origin, pathname } = new URL(document.referrer);
-        if (origin === window.location.origin) {
-          const backBtn = document.createElement('a');
-          backBtn.classList.add('button', 'error-button-back');
-          backBtn.href = pathname;
-          backBtn.textContent = 'Go back';
-          backBtn.title = 'Go back';
-          const btnContainer = document.querySelector('.button-container');
-          btnContainer.append(backBtn);
-        }
-      }
-      sampleRUM('404', { source: document.referrer, target: window.location.href });
-    });
+    import { decorateButtons } from '/scripts/lib-franklin.js';
+    //function to get the html from 404.plain.html
+    async function get404Html() {
+      return fetch('/404.plain.html')
+        .then((response) => {
+          return response.text();
+        })
+        .then((html) => {
+          return html;
+        });
+    }
+
+    get404Html().then(htmlContent => {
```

Comment:
I recommend to make use of `await` for consistency and also use `window.addEventListener('load', () => {` so that we're sure that the DOM is ready before interact with it.

sth like this:

```js
window.addEventListener('load', () => {
  const response = await fetch('/404.plain.html');
  if (response.ok) {
    const htmlContent = await response.text();
    const specialSection = document.querySelector('.link-section');
    specialSection.innerHTML = htmlContent;
    decorateButtons(specialSection);
  }
  sampleRUM('404', { source: document.referrer, target: window.location.href });
});
```